### ISSUE TITLE
IconButton design and documentation updates

### ISSUE THREAD

**Issue by ArakTaiRoth** [GovAlta/ui-components#1211]
h1. Acceptance criteria

* Update icon button active and focus states to align to [Figma design](https://www.figma.com/file/3pb2IK8s2QUqWieH79KdN7/%E2%9D%96-Components-%7C-D3?type=design&node-id=13356%3A390720&t=BVOUMidIiwqek1IL-1)
* Border radius on the icon button and focus ring should be *--goa-borderRadius-m* (it is currently .5rem which is incorrect)
* Icon button icon should *not* show focus (yellow) color on hover/active
* Remove “nocolor” variant from Angular and React code examples
* 
Add note to Properties table for both Angular and React that indicates that the “nocolor” variant is deprecated. Touch base with ? for proper format

Add “*light*” variant, see details in table below

h3. Variant “Color” (default)

||*Screenshot of design*||*Changes needed*||*Change required*||
||{color:1.36B37E}*[ AS DESIGNED ]*{color}|No change|
||{color:1.36B37E}*[ AS DESIGNED ]*{color}|No change|
||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|Icon color should be *--goa-color-interactive-hover*|
||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|Update icon color to *color.interactive.disabled*|

h3. Variant “Light”

||*State*||*Screenshot of design*||*Changes needed*||*Change required*||
|Default||{color:1.6554C0}*[ NEW ]*{color}|Icon *color.greyscale.white*|
|Hover||{color:1.6554C0}*[ NEW ]*{color}|background: *color.greyscale.700*
Icon: *color.greyscale.white*|
|Focus||{color:1.6554C0}*[ NEW ]*{color}|background: *color.greyscale.700*
Icon: *color.greyscale.white*|
|Disabled||{color:1.6554C0}*[ NEW ]*{color}|Icon: *color.greyscale.500*|

h3. Variant “Dark”

||*State*||*Screenshot of design*||*Changes needed*||*Change required*||
|Default||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|Icon *color.greyscale.black*|
|Hover||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|background: *color.greyscale.100*
Icon: *color.greyscale.black*|
|Focus||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|background: *color.greyscale.100*
Icon: *color.greyscale.black*|
|Disabled||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|Icon: *color.greyscale.500*|

h3. Variant “Destructive”

||*State*||*Screenshot of design*||*Changes needed*||*Change required*||
|Default||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|Icon *color.emergency.default*|
|Hover||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|background: *color.emergency.light*
Icon: *color.emergency.dark*|
|Focus||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|background: *color.emergency.light*
Icon: *color.emergency.dark*|
|Disabled||{color:1.FF5630}*[ NEEDS UPDATE ]*{color}|Icon: *1.DA291C 50% opacity*|
